internal/proposals/logs/dynamic-backend-configuration/custom-operator/dynamic-fluentbit-config.md

```diff
@@ -0,0 +1,148 @@
+- Have in mind that the central configuration can be overwritten at any time by reconciliation, which shouldn't reset customers configuration.
+- A new config gets picked up dynamically without any further interaction (like pod restart by the user).
+- Have basic validation to give early feedback on errors -> Debugging a startup problem.
+- OpenTelemetry collector has the same problem to solve thus this proposal should be extensible.
+- There's a way to provide the auth details for a backend config in a secure way.
+
+## Proposal
+
+### Architecture
+
+To build the operator, we use an SDK for a quick and easy start-up, besides various other reasons. For this, we decided to use [`kubebuilder`](https://github.com/kubernetes-sigs/kubebuilder). `kubebuilder` is an upstream project of `Kubernetes`. It provides us with features we need for our operator with the smallest overhead. In addition, this SDK is also used in other parts of Kyma, and thus we've a more consistent picture.
+
+An alternative for `kubebuilder` would be the [`Operator-SDK`](https://github.com/operator-framework/operator-sdk) from `Red Hat`. This has similar features to the kubebuilder, with some additions like the integration of [`operatorhub`](https://operatorhub.io/) - but such features aren't needed for our purpose. Furthermore, this isn't an upstream project of Kubernetes.
+
+To have a simple API, one `CRD` for Fluent Bit configuration is created. This includes a field to determine the type of the configuration (`FILTER` or `OUTPUT`). A list of maps for the Fluent Bit configuration code, and a status field to keep track of the status of the later created CR.
+Using a list of maps for the actual configuration, enables the possibility to support the full Fluent Bit syntax without having the need to maintain new features of various plugins. Furthermore, this gives the users the ability to have a good overview of their sequence of applied filters and outputs. Using the Kyma documentation, we could also lead the users to think more in a way of pipelines, in such that they create one CR for each Fluent Bit pipeline.
+
+One constraint of this operator is, that it doesn't support dynamic plugins, which require to be loaded into the Fluent Bit image. Another constraint is, that not all available plugins for filters and outputs can be configured by this operator (i.e. `Lua`), because such plugins need to have certain files mounted into the container, which isn't possible. If users still want to use such, they need to configure an output plugin of their choice and process the logs with another log processor (i.e. another instance of Fluent Bit). The operator has a list of unsupported plugins, to achieve that such plugins can't be configured. Thus, every time users create a CR, the list is checked if it contains a wanted plugin, if it does users are informed that they aren't able to create such CR.
```

Comment:
What does this sentence mean?
"The operator has a list of unsupported plugins, to achieve that such plugins can't be configured."
As it is, it implies that if the operator didn't have that list, the plugins could be configured; the purpose of the list is to prevent using those plugins. I doubt that's correct. 

Here's my guess:

```suggestion
We're evaluating the following constraints of `kubebuilder`: 
- It doesn't support dynamic plugins, which must be loaded into the Fluent Bit image. 
- Not all available plugins for filters and outputs can be configured by this operator (for example, `Lua`), because such plugins need certain files mounted into the container, which isn't possible. However, this can be mitigated: Users can configure an output plugin of their choice and process the logs with another log processor (for example, another instance of Fluent Bit). If users create a CR, `kubebuilder` checks it against a list of unsupported plugins, and if there is a match, informs the users that they cannot create that CR.
```